tokio/src/sync/semaphore.rs

```diff
@@ -24,6 +24,51 @@ use std::sync::Arc;
 /// To use the `Semaphore` in a poll function, you can use the [`PollSemaphore`]
 /// utility.
 ///
+/// # Examples
+///
+/// ```
+/// use std::sync::Arc;
+/// use tokio::sync::{Semaphore, TryAcquireError};
+///
+/// #[tokio::main]
+/// async fn main() {
+///     let semaphore = Semaphore::new(3);
+///     {
+///         let a_permit = semaphore.acquire().await;
+///         let two_permits = semaphore.acquire_many(2).await;
+///         assert_eq!(a_permit.is_ok(), true);
+///         assert_eq!(two_permits.is_ok(), true);
+///         // all three permits acquired without waiting
+///         assert_eq!(semaphore.available_permits(), 0);
+///         // any more `acquire()` or `acquire_many()` calls will wait
+///         // `try_acquire` attempts to acquire, but fail immediately in this case
+///         let fourth_permit = semaphore.try_acquire();
+///         assert_eq!(fourth_permit.err(), Some(TryAcquireError::NoPermits));
+///         semaphore.close();
+///         // cannot obtain more permits after close
+///         assert_eq!(semaphore.acquire().await.is_err(), true);
+///     } // all permits are dropped at this point
+///     assert_eq!(semaphore.available_permits(), 3);
```

Comment:
I didn't know you could use `available_permits` after closing the semaphore. Anyway, I think it would be simpler to remove the close call from this example. You can have another example that just does a close, or you can have an example on the `close` method only.

Reply:
Yeah that doesn't make sense, I think I left that line by accident. Removed now